java/client/src/main/java/com/vesoft/nebula/client/graph/storage/StorageClient.java

```diff
@@ -194,149 +287,194 @@ private ScanVertexResultIterator scanVertex(String spaceName,
 
 
     /**
-     * scan edge with no specific parts
+     * scan edge of all parts
      *
      * @param spaceName  nebula graph space
+     * @param edgeName   nebula edge name
      * @param returnCols return cols
-     * @param noColumns  if return no col
      * @param limit      return data limit
      * @param startTime  start time
      * @param endTime    end time
      * @return
      */
     public ScanEdgeResultIterator scanEdge(String spaceName,
-                                           Map<String, List<String>> returnCols,
-                                           boolean noColumns,
+                                           String edgeName,
+                                           List<String> returnCols,
                                            int limit,
                                            long startTime,
                                            long endTime) throws Exception {
+        assert (returnCols != null);
+
         Set<Integer> parts = getSpaceParts(spaceName);
+        if (parts.isEmpty()) {
+            throw new TException("No valid part in space " + spaceName);
+        }
+        return scanEdge(spaceName, new ArrayList<>(parts), edgeName,
+                returnCols, false, limit, startTime, endTime);
+    }
+
+
+    /**
+     * scan edge of specific part with no return cols
+     *
+     * @param spaceName nebula graph space
+     * @param edgeName  nebula edge name
+     * @param part      part to scan
+     * @param noColumns return no cols
+     * @param limit     return data limit
+     * @param startTime start time
+     * @param endTime   end time
+     * @return
+     */
+    public ScanEdgeResultIterator scanEdge(String spaceName,
+                                           int part,
+                                           String edgeName,
+                                           boolean noColumns,
+                                           int limit,
+                                           long startTime,
+                                           long endTime) {
+        return scanEdge(spaceName, Arrays.asList(part), edgeName, new ArrayList<>(), noColumns,
+                limit, startTime, endTime);
+    }
 
+
+    /**
+     * scan edge of all parts with no return cols
+     *
+     * @param spaceName nebula graph space
+     * @param edgeName  nebula edge name
+     * @param noColumns return no cols
+     * @param limit     return data limit
+     * @param startTime start time
+     * @param endTime   end time
+     * @return
+     */
+    public ScanEdgeResultIterator scanEdge(String spaceName,
+                                           String edgeName,
+                                           boolean noColumns,
+                                           int limit,
+                                           long startTime,
+                                           long endTime) throws TException {
+
+        Set<Integer> parts = getSpaceParts(spaceName);
```

Comment:
why `getSpaceParts ` return set ?

Reply:
just return the map.keySet, change it to list maybe better